File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-8.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1 

        while l <= r:
            m = (l + r) // 2
            if nums[m] > target:
                if nums[r] > nums[m] or nums[l] <= target:
                    r = m - 1
                else:
                    l = m + 1
            elif nums[m] < target:
                if nums[l] < nums[m] or nums[r] >= target:
                    l = m + 1
                else:
                    r = m - 1
            else:
                return m

            
        return -1

File: Data Structures & Algorithms/top-k-elements-in-list/submission-8.js
class Solution {
    /**
     * @param {number[]} nums
     * @param {number} k
     * @return {number[]}
     */
    topKFrequent(nums, k) {
        const count = {}

        for (const n of nums) {
            count[n] = 1 + (count[n] || 0)
        }
        
        const fq = Array.from( { length: nums.length + 1 }, () => [])
        for (const [key, val] of Object.entries(count)) {
            fq[val].push(key)
        }

        let res = []
        for (let i = nums.length; i > 0; i--) {
            for (const cur of fq[i]) {
                if (k > 0) {
                    res.push(cur)
                    k--
                }
            }
        }
        return res   
    }
}
